### ISSUE TITLE
Client/Server "recursive" calling leads to error

### ISSUE THREAD

**Issue by ris58h** [reactor/reactor-netty#154]
I'm trying to implement [this ratpack fibonacci demo](https://github.com/setblackWs/ratpackFibb/blob/master/src/main/java/pl/setblack/fibbo/FibboServer.java) using reactor-netty.

My code is [here](https://gist.github.com/ris58h/9a3322c7e2989015e3dc09370b42ff7b).
It works on small numbers (up to 8 on my machine) but fails on 9 and greater with "Acquire operation took longer then configured maximum time". [Full stack trace](https://gist.github.com/ris58h/6c95b12a5e97f3c8c0a198f49034f43d). 

Reactor Netty version
0.7.0.M1
JVM version (e.g. `java -version`)
1.8.0_144
OS version (e.g. `uname -a`)
Linux 3.11.0-20-generic x86_64

**Comment by smaldini** [comment]
This seems to be an issue that might have a short solution (default minimum pool connection is 8 where it should be 1000). More long term is that http client should support pipelining to allow such tail recurse situation via a request/response queue.

As a workaround until the default is changed, try doing ```HttpResources.set(PoolResources.fixed("http", 1000)``` before starting calling any thing (static initializer)

**Comment by ris58h** [comment]
@smaldini Thank you for the response. Can I use PoolResources.elastic("http") instead of fixed for a workaround?

**Comment by smaldini** [comment]
Sure that should work as well, usually client caps the number of connections tho. The problem will remain, we want to ultimately smartly limit the number of connections by pipelining http content.

**Comment by violetagg** [comment]
@ris58h PoolResources.elastic is now the default strategy in Reactor Netty.

**Comment by smaldini** [comment]
Closing for now we will specify in a new issue where / how to support http pipelining.
